src/contexts/PositionContext.tsx

```diff
@@ -75,14 +75,17 @@ export const PositionProvider = (props: Props) => {
 
   if (positionId && fetchedPosition?.position) {
     const { position: positionFromTheGraph } = fetchedPosition ?? { position: null }
-
     if (positionFromTheGraph) {
       position = positionFromTheGraph
     } else {
       errors.push(PositionErrors.NOT_FOUND_ERROR)
     }
   }
 
+  if (positionId && !position && !loading) {
+    errors.push(PositionErrors.NOT_FOUND_ERROR)
+  }
+
```

Comment:
The first `if` is checking for a position and then checking if it is null (that is impossible because otherwise, it would never enter that branch).

So lines 76-77 can be written as
```
 if (positionId && fetchedPosition) {
    const { position: positionFromTheGraph } = fetchedPosition
``` 

Another solution is to maintain the new `if` and left the other as

```
if(positionId && fetchedPosition?.position) {
    position = fetchedPosition?.position
}
```

I didn't tested this last alternative but I know that `fetchedPosition` will be undefined (if the query is not loaded) or `{ position: PositionObject | null }`, so the only way that `fetchedPosition?.position` were true is with the object having the PositionObject inside.